File: app/src/main/java/com/duodevloopers/foodup/myapp/MyApp.java
package com.duodevloopers.foodup.myapp;

import android.app.Application;

import com.duodevloopers.foodup.Model.RestaurantItemPojo;
import com.duodevloopers.foodup.Model.User;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;

import java.util.ArrayList;
import java.util.List;

public class MyApp extends Application {

    private static final String TAG = "MyApp";

    public static List<RestaurantItemPojo> mRestaurantItemPojo = new ArrayList<>();

    private static User user;

    private static void getUser() {
        FirebaseFirestore.getInstance()
                .collection("users")
                .document(FirebaseAuth.getInstance().getCurrentUser().getPhoneNumber())
                .get()
                .addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
                    @Override
                    public void onSuccess(DocumentSnapshot documentSnapshot) {
                        if (documentSnapshot.exists()) {
                            user = documentSnapshot.toObject(User.class);
                        }
                    }
                });
    }

    public static User getLoggedInUser() {
        return user;
    }

    public static void addItemFromCart(RestaurantItemPojo restaurantItemPojo) {

        List<RestaurantItemPojo> toRemove = new ArrayList<>();

        for (RestaurantItemPojo item : mRestaurantItemPojo) {
            if (item.getmFoodName().equals(restaurantItemPojo.getmFoodName())) {
                toRemove.add(item);
            }
        }

        mRestaurantItemPojo.removeAll(toRemove);
        mRestaurantItemPojo.add(restaurantItemPojo);
    }

    public static void addItem(RestaurantItemPojo restaurantItemPojo) {

        if (mRestaurantItemPojo.contains(restaurantItemPojo)) {
            mRestaurantItemPojo.remove(restaurantItemPojo);
        } else {
            mRestaurantItemPojo.add(restaurantItemPojo);
        }

    }

    @Override
    public void onCreate() {
        super.onCreate();

        if (FirebaseAuth.getInstance().getCurrentUser() != null) {
            //getUser();
        }
        user = new User();
        user.id = "C171080";
        user.credit = "300.00";
        user.section = "1CM";

    }

    public static void removeItem(RestaurantItemPojo restaurantItemPojo) {
        mRestaurantItemPojo.remove(restaurantItemPojo);
    }

    public static List<RestaurantItemPojo> getmRestaurantItemPojo() {
        return mRestaurantItemPojo;
    }
}

File: app/src/main/java/com/duodevloopers/foodup/myapp/MyApp.kt
package com.duodevloopers.foodup.myapp

import android.app.Application
import android.util.Log
import com.duodevloopers.foodup.Model.RestaurantItemPojo
import com.duodevloopers.foodup.Model.User
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore
import java.util.*
import kotlin.collections.ArrayList

class MyApp : Application() {
    override fun onCreate() {
        super.onCreate()
        if (FirebaseAuth.getInstance().currentUser != null) {
            getUser()
            getPrintShops()
        }

    }


    companion object {
        private const val TAG = "MyApp"
        var mRestaurantItemPojo: MutableList<RestaurantItemPojo> = ArrayList()
        var loggedInUser: User? = null
        var shopList: MutableList<String> = ArrayList()

        private fun getUser() {

            val ref = FirebaseFirestore.getInstance().collection("student")
                .document(FirebaseAuth.getInstance().currentUser!!.phoneNumber!!)

            ref.addSnapshotListener { snapshot, e ->
                if (e != null) {
                    Log.w(TAG, "Listen failed.", e)
                    return@addSnapshotListener
                }

                if (snapshot != null && snapshot.exists()) {

                    loggedInUser = snapshot.toObject(
                        User::class.java
                    )

                } else {
                    Log.d(TAG, "Current data: null")
                }
            }
        }

         fun getPrintShops() {
            FirebaseFirestore.getInstance()
                .collection("shops")
                .get()
                .addOnSuccessListener {
                    for (shop in it.documents) {

                        if (shop["type"] != "food") shopList.add(shop["name"].toString())

                    }
                }
        }

        @JvmStatic
        fun addItemFromCart(restaurantItemPojo: RestaurantItemPojo) {
            val toRemove: MutableList<RestaurantItemPojo> = ArrayList()
            for (item in mRestaurantItemPojo) {
                if (item.getmFoodName() == restaurantItemPojo.getmFoodName()) {
                    toRemove.add(item)
                }
            }
            mRestaurantItemPojo.removeAll(toRemove)
            mRestaurantItemPojo.add(restaurantItemPojo)
        }

        @JvmStatic
        fun addItem(restaurantItemPojo: RestaurantItemPojo) {
            if (mRestaurantItemPojo.contains(restaurantItemPojo)) {
                mRestaurantItemPojo.remove(restaurantItemPojo)
            } else {
                mRestaurantItemPojo.add(restaurantItemPojo)
            }
        }

        fun removeItem(restaurantItemPojo: RestaurantItemPojo) {
            mRestaurantItemPojo.remove(restaurantItemPojo)
        }

        @JvmStatic
        fun getmRestaurantItemPojo(): List<RestaurantItemPojo> {
            return mRestaurantItemPojo
        }
    }

}